src/emulator/storage/files.ts

```diff
@@ -89,14 +86,21 @@ export type ListObjectsRequest = {
   authorization?: string;
 };
 
-/**  Parsed request object for {@link StorageLayer#handleCreateDownloadToken}. */
+/** Response object for {@link StorageLayer#listObjects}. */
+export type ListObjectsResponse = {
```

Comment:
I'd combine this and the existing [ListResponse](https://github.com/firebase/firebase-tools/blob/ba53736ade37a4bc147fc792d2403fe1e4635149/src/emulator/storage/list.ts#L10) type. If you choose to return undefined instead of empty arrays (as is the current behaviour), make sure this doesn't break anything on the SDK side.

Reply:
Removed `list.ts`, added another integration test to make sure the response shape hasn't changed.

Reply:
Could you make the new type have non-optional properties then? Then no need for `?? []`s

Reply:
? In `gcloud.ts` is it not just `listResponse.items.map(...)`?

Reply:
We expect the `items` and `prefix` key to not be present in the response object of `gcloud.ts` if there are no elements in those arrays. I'm not sure express strips out empty arrays passed into `res.json`

Reply:
Talked about this offline, what you have now is fine then